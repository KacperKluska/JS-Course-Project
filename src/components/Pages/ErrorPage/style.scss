.errorScreen {
  display: flex;
  flex-direction: column;
  align-items: flex-start;
  justify-content: flex-start;

  min-height: calc(
    100vh - 50px - 80px - 300px
  ); //50px - navbar, 80px - header, +/- 300px - footer
  width: 100%;

  background-color: transparent;

  padding: 2rem;

  h1::after {
    content: "";
    background-color: black;
    height: 1px;
    display: block;
    margin-bottom: 1rem;
  }
}

.linkA {
  color: black;
}
.linkA:hover {
  color: blue;
  text-decoration: underline;
}
