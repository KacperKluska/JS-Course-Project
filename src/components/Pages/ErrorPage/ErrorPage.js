import React from 'react';
import { Link, useLocation } from 'react-router-dom';
import './style.scss';

const ErrorPage = ({ errorCode, children }) => {
  const location = useLocation();

  return (
    <div className="errorScreen">
      <h1>{errorCode}</h1>
      <p>
        Couldn&apos;t find localhost{location.pathname}
        <br />
        {children}
        <br />
        Click{' '}
        <Link className="linkA" to="/">
          <b>here</b>
        </Link>{' '}
        to go back to main page
      </p>
    </div>
  );
};

export default ErrorPage;
