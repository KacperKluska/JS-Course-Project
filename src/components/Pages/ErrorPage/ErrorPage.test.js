import { BrowserRouter } from "react-router-dom";
import ErrorPage from "./ErrorPage";

let wrapper;

test("ErrorPage snapshot", () => {
  wrapper = mount(
    <BrowserRouter>
      <ErrorPage errorCode="404 page not found">
        This page doesn't exist. Try to go back.
      </ErrorPage>
    </BrowserRouter>
  );
  expect(wrapper).toMatchSnapshot();
});
