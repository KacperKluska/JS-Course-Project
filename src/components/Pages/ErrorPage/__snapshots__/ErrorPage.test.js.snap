// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorPage snapshot 1`] = `
<div
  className="errorScreen"
>
  <h1>
    404 page not found
  </h1>
  <p>
    Couldn't find localhost
    /
    <br />
    This page doesn't exist. Try to go back.
    <br />
    Click
     
    <a
      className="linkA"
      href="/"
      onClick={[Function]}
    >
      <b>
        here
      </b>
    </a>
     
    to go back to main page
  </p>
</div>
`;
